synapse/handlers/room_member.py

```diff
@@ -1581,7 +1581,11 @@ async def _remote_join(
         ]
 
         if len(remote_room_hosts) == 0:
-            raise SynapseError(404, "No known servers")
+            raise SynapseError(
+                404,
+                "Can't join remote room because no servers "
+                "that are in the room have been provided.",
+            )
```

Comment:
Should this be 400 rather than 404? Would be nice to have a matrix error code too

Reply:
I really really want to, but I don't want to interfere with it this far down the call tree because I don't know how that looks like for other endpoints that call here.
(Ideally I think a `400` would be in order if you try to join a **room ID** with no server_name, but what about if the server is already joined — should it be allowed then?)

But what about if you resolve a *room alias* and wind up with no servers to join by (is that possible? does 400 make sense there?).

The description changes felt much less harmless but I'd have to actually think about changing the codes, I think :/

Reply:
Very fair, entirely reasonable, thoroughly considered.

I think the take away is "let's not raise HTTP status codes from the bowels of the application"?